compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -1609,17 +1609,7 @@ pub type PlaceholderType = Placeholder<BoundVar>;
 pub type PlaceholderConst = Placeholder<BoundVar>;
 
 /// A `DefId` which is potentially bundled with its corresponding generic parameter
-/// in case `did` is a const argument.
-///
-/// This is used to prevent cycle errors during typeck
-/// as `type_of(const_arg)` depends on `typeck(owning_body)`
-/// which once again requires the type of its generic arguments.
-///
-/// Luckily we only need to deal with const arguments once we
-/// know their corresponding parameters. We (ab)use this by
-/// calling `type_of(param_did)` for these arguments.
-///
-/// # Examples
+/// in case `did` is a const argument. Used to prevent cycle errors during `typeck`.
```

Comment:
In the example below, what thing is this the DefId of? The "7" const argument?

Reply:
It is used for everything, regardless of if it is a const argument. Mentioned at the end of the docs. Would it make sense to move that section upwards? Is that section even useful to you?
```rust
/// As we share the same code paths between const args and const params we
/// encapsulate this using `WithOptConstParam` and additional
/// `query_name_opt_const_arg` methods, which delegate to the correct query
/// depending on the value of `const_param_did`.
```

Reply:
Sure it is used for everything, but what I meant was -- where is the connection between the problem described below, and the concrete type this is a doc comment for? The example should be related to  `WithOptConstParam`, by giving a concrete element of that type and talking about it.

(Also I wonder if there is a more descriptive name than "with optional const parameter", which says basically nothing... something that describes the intent of this, not just its structure.)